nautobot/ipam/views.py

```diff
@@ -316,92 +315,6 @@ class AggregateBulkDeleteView(generic.BulkDeleteView):
     table = tables.AggregateTable
 
 
-#
-# Prefix/VLAN roles
-#
-
-
-class RoleListView(generic.ObjectListView):
-    queryset = Role.objects.annotate(
-        prefix_count=count_related(Prefix, "role"),
-        vlan_count=count_related(VLAN, "role"),
-    )
-    filterset = filters.RoleFilterSet
-    table = tables.RoleTable
-
-
-class RoleView(generic.ObjectView):
-    queryset = Role.objects.all()
-
-    def get_extra_context(self, request, instance):
-
-        # Prefixes
-        # v2 TODO(jathan): Replace prefetch_related with select_related
-        prefixes = (
-            Prefix.objects.restrict(request.user, "view")
-            .filter(role=instance)
-            .prefetch_related(
-                "site",
-                "status",
-                "tenant",
-                "vlan",
-                "vrf",
-            )
-        )
-
-        prefix_table = tables.PrefixTable(prefixes)
-        prefix_table.columns.hide("role")
-
-        paginate = {
-            "paginator_class": EnhancedPaginator,
-            "per_page": get_paginate_count(request),
-        }
-        RequestConfig(request, paginate).configure(prefix_table)
-
-        # VLANs
-        # v2 TODO(jathan): Replace prefetch_related with select_related
-        vlans = (
-            VLAN.objects.restrict(request.user, "view")
-            .filter(role=instance)
-            .prefetch_related(
-                "group",
-                "site",
-                "status",
-                "tenant",
-            )
-        )
-
-        vlan_table = tables.VLANTable(vlans)
-        vlan_table.columns.hide("role")
-
-        RequestConfig(request, paginate).configure(vlan_table)
-
-        return {
-            "prefix_table": prefix_table,
-            "vlan_table": vlan_table,
-        }
```

Comment:
Some net loss of UI functionality here - the new generic Role detail view doesn't show prefixes and vlans that are using this role. Might be nice to keep that functionality and extend it to include devices/racks/ipaddresses as well?